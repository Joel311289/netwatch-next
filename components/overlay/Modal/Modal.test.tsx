import { act, render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Modal from '@/components/overlay/Modal/Modal';

const portalContent = 'Content';

const content = () => <div>{portalContent}</div>;

describe('Tests Modal component', () => {
  test('renders component correctly', () => {
    const { container } = render(<Modal opened>Test</Modal>);

    expect(container).toMatchSnapshot();
  });

  test('renders children correctly', () => {
    render(<Modal opened>{content()}</Modal>);

    expect(screen.getByText(portalContent)).toBeInTheDocument();
  });

  test('closes portal correctly', async () => {
    render(<Modal opened>{content()}</Modal>);

    expect(screen.getByText(portalContent)).toBeInTheDocument();

    // eslint-disable-next-line testing-library/no-unnecessary-act
    act(() => {
      userEvent.click(screen.getByLabelText('close'));
    });

    await waitFor(async () => {
      expect(screen.queryByText(portalContent)).not.toBeInTheDocument();
    });
  });
});
