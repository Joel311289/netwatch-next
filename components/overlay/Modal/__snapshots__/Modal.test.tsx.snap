// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests Modal component renders component correctly 1`] = `
<div
  aria-hidden="true"
/>
`;
